### PR TITLE
Update ruy to the latest upstream commit.

### LINKED ISSUE #3428: Update to the latest version of ruy
The submodule is 6mo+ old and we are hitting the generic CPU path.

### PR DESCRIPTION
**Pull request by bjacob**
Rewrite CMakeLists.txt to actually do the correct SIMD flag-setting as
in the upstream BUILD, and to be more correct e.g. declaring all
dependencies even on header-only internal libraries.

The immediate motivation for this is enabling x86 SIMD (AVX/AVX2/AVX512)
code paths.

Closes #3428

### CODE REVIEW

**Comment by hanhanW** [comment]
fyi, there is an option to convert it to draft.
image

**Comment by benvanik** [comment]
In ruy's kernel_x86.h the define is `_MSC_VER` (not `__MSC_VER`); like:
`#if (defined __clang__) || (defined _MSC_VER)`

File: build_tools/third_party/ruy/CMakeLists.txt [hunk 1]
```diff
@@ -12,25 +12,72 @@
...
+  set(RUY_COPTS_NEON -mfpu=neon)
+endif()
+
+set(RUY_COPTS
+  ${RUY_COPTS_BASE}
+  ${RUY_COPTS_WARNINGS}
+  ${RUY_COPTS_NEON}
+)
+
+if(CMAKE_SYSTEM_PROCESSOR STREQUAL x86_64 OR
```
Comment (benvanik) [inline build_tools/third_party/ruy/CMakeLists.txt:50]: apparently it's AMD64 (sometimes); this seems to be the recommended way to handle it:
```
string(TOLOWER "${CMAKE_SYSTEM_PROCESSOR}" _CMAKE_SYSTEM_PROCESSOR_LOWER)
if(_CMAKE_SYSTEM_PROCESSOR_LOWER STREQUAL x86_64 OR
   _CMAKE_SYSTEM_PROCESSOR_LOWER STREQUAL amd64)
```

**Review by benvanik — CHANGES_REQUESTED** [review]
two minor fixes (one upstream ruy, one here) then it seems to work for me on windows/msvc/avx2!

**Review by benvanik — APPROVED** [review]

**Pull request merged**